### PR TITLE
chore: set scroll position on zoom change

### PR DESCRIPTION
**Pull request by AyushMittal1999**

### CODE REVIEW

File: src/PanAndZoomContainer.js [hunk 1]
```diff
@@ -59,17 +54,44 @@ const PanAndZoomContainer = ({
...
+    diagramContainerRef.current.style.transform = `translate3D(${x}px, ${y}px, 0) scale(${zoom})`;
+
+    container.scrollLeft = containerRef.current.scrollLeft =
+      (scroll.left * zoom) / previousZoom.current +
+      ((zoom - previousZoom.current) * containerWidth) /
+        (2 * previousZoom.current);
+    container.scrollTop =
+      (scroll.top * zoom) / previousZoom.current +
+      (scroll.top
+        ? ((zoom - previousZoom.current) * containerHeight) /
```
Comment (ujjawalpabreja99) [inline src/PanAndZoomContainer.js:84]: why do we need this conditional check?

File: src/PanAndZoomContainer.js [hunk 2]
```diff
@@ -59,17 +54,44 @@ const PanAndZoomContainer = ({
...
+      y = ((containerHeight - extremeY) * zoom) / 2;
+    }
+
+    diagramContainerRef.current.style.transform = `translate3D(${x}px, ${y}px, 0) scale(${zoom})`;
+
+    container.scrollLeft = containerRef.current.scrollLeft =
+      (scroll.left * zoom) / previousZoom.current +
+      ((zoom - previousZoom.current) * containerWidth) /
+        (2 * previousZoom.current);
+    container.scrollTop =
```
Comment (ujjawalpabreja99) [inline src/PanAndZoomContainer.js:81]: `containerRef.current.scrollTop` not updated here? As done on [line 77](https://github.com/ujjawalpabreja99/react-virtualized-flowchart/pull/2/files#diff-fa50ff659ac0d0bad829809f3061f7a785c9852fd029b4dacad4215138058323R77)

File: src/PanAndZoomContainer.js [hunk 3]
```diff
@@ -59,17 +54,44 @@ const PanAndZoomContainer = ({
...
+      (scroll.top * zoom) / previousZoom.current +
+      (scroll.top
+        ? ((zoom - previousZoom.current) * containerHeight) /
+          (2 * previousZoom.current)
+        : 0);
+
     previousZoom.current = zoom;
-  }, [zoom]);
+  }, [containerRef, contentSpan.x, contentSpan.y, pan.x, pan.y, zoom]);
```
Comment (ujjawalpabreja99) [inline src/PanAndZoomContainer.js:89]: i doubt if we need `containerRef` as a dependency ([reference](https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/))

File: src/Diagram.js [hunk 4]
```diff
@@ -279,6 +283,7 @@ const Diagram = props => {
         containerRef={containerRef}
         renderPanAndZoomControls={props.renderPanAndZoomControls}
         scroll={state.scroll}
+        contentSpan={{ x: extremeX, y: extremeY }}
```
Comment (ujjawalpabreja99) [inline src/Diagram.js:286]: can memoize `contentSpan` ?

Comment (AyushMittal1999) [reply src/Diagram.js:286]: I think there is no need, as we are dependent upon x and y values directly in [PanAndZoomContainer](https://github.com/ujjawalpabreja99/react-virtualized-flowchart/pull/2/files#diff-fa50ff659ac0d0bad829809f3061f7a785c9852fd029b4dacad4215138058323)

File: src/PanAndZoomContainer.js (hunk 1 above)
Comment (AyushMittal1999) [reply src/PanAndZoomContainer.js:84]: if scroll.top is zero, then we don't not to move vertical scroll.
For eg: If we are in 20% zoom, we dont want to scroll the diagram vertically unless user scroll a little bit.

File: src/Diagram.js (hunk 4 above)
Comment (AyushMittal1999) [reply src/Diagram.js:286]: No need to memoize, we are dependent upon values of `x` and `y` only in underlying component.

File: src/PanAndZoomContainer.js (hunk 3 above)
Comment (AyushMittal1999) [reply src/PanAndZoomContainer.js:89]: We didn't need here.
Actually, `containerRef` is coming as a prop in the component, so just for some edge-case, I have added.

File: src/PanAndZoomContainer.js (hunk 2 above)
Comment (AyushMittal1999) [reply src/PanAndZoomContainer.js:81]: line 77 was a typo updated now.

File: src/PanAndZoomContainer.js (hunk 1 above)
Comment (ujjawalpabreja99) [reply src/PanAndZoomContainer.js:84]: will that scenario also occur for `scroll.left` ?

Comment (AyushMittal1999) [reply src/PanAndZoomContainer.js:84]: Yes, but for horizontal, we intentionally want to scroll the diagram keeping it horizontally centered

**Pull request merged**